File: src/main/cpp/addon/AddonEvent.cpp
#include "include/AddonEvent.h"

namespace ravex {
namespace addon {

AddonEvent::AddonEvent(const std::string& name) : name(name), cancelled(false) {}
std::string AddonEvent::getName() const { return name; }
bool AddonEvent::isCancelled() const { return cancelled; }
void AddonEvent::setCancelled(bool cancel) { cancelled = cancel; }

}
}

File: src/main/java/ravex/addon/event/AddonEvent.java
package ravex.addon.event;

public class AddonEvent {
    private final String name;
    private boolean cancelled = false;

    public AddonEvent(String name) {
        this.name = name;
    }

    public String getName() { return name; }
    public boolean isCancelled() { return cancelled; }
    public void setCancelled(boolean cancelled) { this.cancelled = cancelled; }
}
